### ISSUE TITLE
Unsaved file prompt not keyboard navigable on Linux

### ISSUE THREAD

**Issue by suhaibkn** [zed-industries/zed#15151]
### Check for existing issues

- [X] Completed

### Describe the feature

Upon closing a file, if the file had unsaved changed and the unsaved file prompt turns up, we need to move the mouse and click an action to register actions on this prompt.

Ideally, we should be able to select different action buttons using the left/right (or h/l) keys, and then hit Return to trigger the selected action.

Ubuntu 24.04 (x86-64)
Wayland, on GNOME
Zed Preview 0.146.0

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/user-attachments/assets/5e139004-c05f-4890-bcf8-329084fc3558)
